build.rs

```diff
@@ -83,6 +83,7 @@ fn main() {
         println!(
             "cargo:rustc-check-cfg=cfg(__ZEROCOPY_INTERNAL_USE_ONLY_NIGHTLY_FEATURES_IN_TESTS)"
         );
+        println!("cargo:rustc-check-cfg=cfg(zerocopy_unstable)");
```

Comment:
How do we make sure we test both w/ and w/o this cfg in CI without duplicating every CI test?

Reply:
I think our options here are to either:
- test on all our CI targets and eat the resource cost
- test on a subset of CI and eat the risk
- maintain a contract of `zerocopy_unstable` only being additive
- don't use zerocopy_unstable and use `doc(hidden)` instead

If our main motivation for landing `KnownLayout::MaybeUninit` is unblocking a generic value type, then I'm inclined to say we should be using `doc(hidden)`, because that API would subsume `MaybeUninit` *and* `UnalignUnsized`.

Reply:
Discussed offline. Will land in 0.9, upgrading the existing `Unalign` type to support unsized types rather than adding a new `UnalignUnsized`.